pkg/sql/create_tenant.go

```diff
@@ -0,0 +1,37 @@
+// Copyright 2022 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package sql
+
+import (
+	"context"
+
+	"github.com/cockroachdb/cockroach/pkg/roachpb"
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
+)
+
+type createTenantNode struct {
+	name roachpb.TenantName
+}
+
+func (p *planner) CreateTenantNode(_ context.Context, n *tree.CreateTenant) (planNode, error) {
```

Comment:
wondering, is this where we should verify things like name uniqueness and maybe permissions? or should we rely on startExec to fail?

Reply:
- name uniqueness is ensured via the UNIQUE index on the column.
- permission is ensured inside the `CreateTenant` function.